selfservice/strategy/code/strategy_registration.go

```diff
@@ -253,7 +251,7 @@ func (s *Strategy) registrationVerifyCode(ctx context.Context, f *registration.F
 
 	// Step 2: Check if the flow traits match the identity traits
 	for _, n := range container.NewFromJSON("", node.DefaultGroup, p.Traits, "traits").Nodes {
-		if !strings.EqualFold(f.GetUI().GetNodes().Find(n.ID()).Attributes.GetValue().(string), n.Attributes.GetValue().(string)) {
+		if f.GetUI().GetNodes().Find(n.ID()).Attributes.GetValue() != n.Attributes.GetValue() {
```

Comment:
What if we now have strings that have different casing? I think we'll need a type switch here.